app/src/main/java/com/example/joshuarodriguez/mvpcalculator/mvp/mvp/presenter/CalculatorPresenter.java

```diff
@@ -20,19 +23,64 @@
     public CalculatorPresenter(CalculatorModel calculatorModel, CalculatorView calculatorView) {
         this.calculatorModel = calculatorModel;
         this.calculatorView = calculatorView;
+        init();
+    }
+
+    private void init() {
+        AppCompatActivity activity = calculatorView.getActivity();
```

Comment:
refactorize to
`

AppCompatActivity activity = calculatorView.getActivity();
 if (activity != null) {
      return;
}
    setPortraitOrientation(activity);
calculatorView.hideKeyboard();

`